dotnet/src/SemanticKernel/Orchestration/SKFunction.cs

```diff
@@ -209,7 +209,7 @@ public ISKFunction SetAIService(Func<ITextCompletion> serviceFactory)
     {
         Verify.NotNull(serviceFactory, "AI LLM service factory is empty");
         this.VerifyIsSemantic();
-        this._aiService = serviceFactory.Invoke();
+        this._aiService = new Lazy<ITextCompletion>(serviceFactory);
```

Comment:
@dluc Please take a closer look here... this was setting as a func() so that it could be delay-loaded, but then it was invoking it immediately anyway, upon semantic function registration. One effect was that, if you registered functions before AI connectors, it would throw.

Note: even with the "Lazy" approach, if you change your default completions service after or between calls to the function, the change will not be reflected. Another approach would be to store this._aiServiceFactory here and only construct/dispose upon each call to the function.

Reply:
Update: I see from ItFailsIfTextCompletionServiceConfigIsNotSet that it's expected to throw if you register a function before a completion service. Do we want to stick with that (even though you could feasibly remove the completion service right after registering the function)?